File: merge_sort.py
def merge_sort(arr):
    if len(arr) <= 1:
        return arr
    
    center = len(arr) // 2
    left_part = arr[:center]
    right_part = arr[center:]
    
    left_part = merge_sort(left_part)
    right_part = merge_sort(right_part)
    
    return list(merge(left_part, right_part))





def merge(left_part, right_part):
    res = []
    
    while len(left_part) != 0 and len(right_part) != 0:
        if left_part[0] < right_part[0]:
            res.append(left_part[0])
            left_part.remove(left_part[0])
        else:
            res.append(right_part[0])
            right_part.remove(right_part[0])
    
    if len(left_part) == 0:
        res += right_part
    else:
        res += left_part
    
    return res




array = [12, 34, 25, 15, 67, 23, 11, 5, 86]
print("Initial array: ", array)

result_array = merge_sort(array)
print("Result array: ", result_array)

File: Merge Sort/MergeSort.cs
public List<double> MergeSort(List<double> list)
        {
            if (list.Count <= 1)
            {
                return list;
            }

            var center = list.Count / 2;
            
            var leftPart = list.GetRange(0, center);
            var rightPart = list.GetRange(center, list.Count - center);

            leftPart = MergeSort(leftPart); 
            rightPart = MergeSort(rightPart);

            return Merge(leftPart, rightPart);
        }

private static List<double> Merge(List<double> leftPart, List<double> rightPart)
{
    var resultList = new List<double>();
    
    while (leftPart.Count != 0 && rightPart.Count != 0)
    {
        if (leftPart[0] < rightPart[0])
        {
            resultList.Add(leftPart[0]);
            leftPart.Remove(leftPart[0]);
        }
        else
        {
            resultList.Add(rightPart[0]);
            rightPart.Remove(rightPart[0]);
        }
    }

    if (leftPart.Count == 0)
    {
        for (var i = 0; i < rightPart.Count; i++)
        {
            resultList.Add(rightPart[i]);
        }
    }
    else
    {
        for (var i = 0; i < leftPart.Count; i++)
        {
            resultList.Add(leftPart[i]);
        }
    }

    return resultList;
}